src/main/java/executor/service/service/parallelflowexecutor/impls/subscribers/DefaultPairGeneratorService.java

```diff
@@ -0,0 +1,78 @@
+package executor.service.service.parallelflowexecutor.impls.subscribers;
+
+import executor.service.model.proxy.ProxyConfigHolder;
+import executor.service.model.scenario.Scenario;
+import executor.service.queue.ProxyQueue;
+import executor.service.queue.ScenarioQueue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DefaultPairGeneratorService implements PairGeneratorService {
+    private final ProxyQueue proxyQueue;
+    private final ScenarioQueue scenarioQueue;
+
+    private final ProxyConfigHolder defaultProxy;
+
+    public DefaultPairGeneratorService(ProxyQueue proxyQueue, ScenarioQueue scenarioQueue, ProxyConfigHolder defaultProxy) {
+        this.proxyQueue = proxyQueue;
+        this.scenarioQueue = scenarioQueue;
+        this.defaultProxy = defaultProxy;
+    }
+
+
+    @Override
+    public List<Pair> generatePairs() {
+        List<Pair> pairs = new ArrayList<>();
+        List<ProxyConfigHolder> proxies;
+        List<Scenario> scenarios;
+        try {
+            proxies = proxyQueue.getAllProxy();
+            scenarios = scenarioQueue.getAllScenario();
+            if (!scenarios.isEmpty() && !proxies.isEmpty()) {
+                if (proxies.size() == scenarios.size()) {
+                    pairs = createOneToOnePairs(proxies, scenarios);
+                } else {
+                    pairs = createMixPairs(proxies, scenarios);
+                }
```

Comment:
better to hide this if/else in one function with common signature, and like this (but try to find out better name) :
return combineProxiesAndScenarios(proxies, scenario);